packages/block-library/src/template-part/edit/import-controls.js

```diff
@@ -77,10 +76,10 @@ export function TemplatePartImportControls( { area, setAttributes } ) {
 		];
 	}, [ sidebars ] );
 
-	// Only bailout from rendering after data is loaded, and there is nothing to import,
+	// Render an empty node while data is loading or when there is nothing to import;
 	// to avoid SlotFill re-positioning issue. See: https://github.com/WordPress/gutenberg/issues/15641.
-	if ( hasResolved && ! options.length ) {
-		return null;
+	if ( ! hasResolved || ( hasResolved && ! options.length ) ) {
```

Comment:
I'm fine with it as is, but we could have two conditions here.
1. if (!hasResolved) return the Spacer
2. if hasResolved && !options) return null

Reply:
Good point. Update the logic.